ci/cron/src/Main.hs

```diff
@@ -0,0 +1,254 @@
+        JSON.object ["name" JSON..= ("Release of DAML SDK " <> version),
+                     "post_body" JSON..= body,
+                     "content_group_id" JSON..= (11411412838 :: Integer),
+                     "publish_date" JSON..= date,
+                     "post_summary" JSON..= summary,
+                     "slug" JSON..= version,
+                     "blog_author_id" JSON..= (11513309969 :: Integer),
+                     "meta_description" JSON..= summary]
+
+tell_hubspot :: GitHubVersion -> IO ()
+tell_hubspot latest = do
+    putStrLn $ "Publishing "<> (name latest) <> " to Hubspot..."
+    desc <- http_post "https://api.github.com/markdown/raw" [("Content-Type", "text/plain")] (LBS.fromString $ notes latest)
+    date <- read <$> (<> "000") <$> init <$> (shell $ "date -d " <> (published_at latest) <> " +%s")
+    let summary = "Release notes for version " <> (name latest) <> "."
+    token <- Env.getEnv "HUBSPOT_TOKEN"
+    submit_blog <- http_post ("https://api.hubapi.com/content/api/v2/blog-posts?hapikey=" <> token)
+                             [("Content-Type", "application.json")]
+                             (JSON.encode $ SubmitBlog { body = LBS.toString desc,
+                                                         date,
+                                                         summary,
+                                                         version = (name latest) })
+    case JSON.decode submit_blog of
+      Nothing -> Exit.die $ "No blog id from HubSpot: \n" <> LBS.toString submit_blog
+      Just (BlogId { blog_id }) -> do
+          _ <- http_post ("https://api.hubapi.com/content/api/v2/blog-posts/" <> show blog_id <> "/publish-action?hapikey=" <> token)
+                         [("Content-Type", "application.json")]
+                         (JSON.encode $ JSON.object [("action", "schedule-publish")])
+          return ()
+
+data GitHubVersion = GitHubVersion { prerelease :: Bool, tag_name :: String, notes :: String, published_at :: String } deriving Show
+instance JSON.FromJSON GitHubVersion where
+    parseJSON = JSON.withObject "GitHubVersion" $ \v -> GitHubVersion
+        <$> v JSON..: (Text.pack "prerelease")
+        <*> v JSON..: (Text.pack "tag_name")
+        <*> v JSON..:? (Text.pack "body") JSON..!= ""
+        <*> v JSON..: (Text.pack "published_at")
+
+name :: GitHubVersion -> String
+name gh = tail $ tag_name gh
+
+main :: IO ()
+main = do
+    robustly_download_nix_packages
+    putStrLn "Checking for new version..."
+    gh_resp <- remove_prereleases <$> http_get "https://api.github.com/repos/digital-asset/daml/releases"
+    docs_resp <- docs_versions <$> http_get "https://docs.daml.com/versions.json"
+    if (github_versions gh_resp) == docs_resp
+    then do
+        putStrLn "No new version found, skipping."
+        Exit.exitSuccess
+    else do
+        Temp.withSystemTempDirectory "docs" $ \docs_folder -> do
```

Comment:
Afaik, we mostly use the functionality from `extra` in other places, i.e., `withTempDir` from `System.IO.Extra` so I would recommend to also use this here.

Reply:
How would it be different? What would it buy me in this code? Is this not working as is?